src/inapp/inapp.ts

```diff
@@ -347,37 +347,30 @@ export function getInAppMessages(
           }
 
           /* 
-            track in-app consumes only when _saveToInbox_ 
-            is falsy or undefined and always track in-app opens
+            track in-app consumes only when _trigger.type_ 
+            is NOT "never" or undefined and always track in-app opens
 
             Also swallow any 400+ response errors. We don't care about them.
           */
           if (ENABLE_INAPP_CONSUME || IS_PRODUCTION) {
-            Promise.all(
-              !activeMessage?.saveToInbox
-                ? [
-                    trackInAppOpen({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    }),
-                    trackInAppConsume({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    })
-                  ]
-                : [
-                    trackInAppOpen({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    })
-                  ]
-            ).catch((e) => e);
+            const trackRequests = [
+              trackInAppOpen({
+                messageId: activeMessage.messageId,
+                deviceInfo: {
+                  appPackageName: payload.packageName
+                }
+              })
+            ];
+            if (activeMessage.trigger.type !== 'never')
+              trackRequests.push(
```

Comment:
Opting to push `trackInAppConsume` to the `trackRequests` array to reduce duplicate code.